dataprofiler/data_readers/data_utils.py

```diff
@@ -303,6 +302,47 @@ def detect_file_encoding(file_path, buffer_size=1024, max_lines=20):
     # Typical file representation is utf-8 instead of ascii, treat as such.
     if not encoding or encoding == 'ascii':
         encoding = 'utf-8'
+
+    # Check if encoding can be used to decode without throwing an error
+    def _decode_is_valid(encoding):
+        try: 
+            open(file_path, "rb").read().decode(encoding.lower())
+            return True
+        except: return False
+
+    if not _decode_is_valid(encoding):
+        try:
+            from charset_normalizer import CharsetNormalizerMatches as CnM
+            
+            # Try with small sample 
+            with open(file_path, 'rb') as input_file:
+                raw_data = input_file.read(2560)
```

Comment:
How was that selected?

Reply:
@lettergram In your code block you had tested for the small sample with

rawdata = in_file.read(10000)
...
results = CnM.from_bytes(
    rawdata,
    steps=5,  # Number of steps/block to extract from my_byte_str
    chunk_size=512,  # Set block size of each extraction
...

5 steps * 512 byte chunk size = 2560 bytes? Which is why I only read in 2560 bytes. Should I change this to do 10k?

Reply:
Interesting, I'm not sure how it works, I recall I was getting less accurate results when I read in smaller amounts.

I'm fine leaving it, but was curious.

Reply:
I'd read in 10k samples, above you're already reading the entire file